include/zephyr/drivers/spi.h

```diff
@@ -681,12 +735,13 @@ static inline int spi_transceive_async(const struct device *dev,
 				       const struct spi_config *config,
 				       const struct spi_buf_set *tx_bufs,
 				       const struct spi_buf_set *rx_bufs,
-				       struct k_poll_signal *async)
+				       struct k_poll_signal *sig)
```

Comment:
you'll need to adapt the documentation.

Actually, I think the function should be renamed to spi_transceive_sig, so just create a helper called spi_transceive_async() (pointing back to spi_transceive_sig) that you mark as deprecated then.